### PR TITLE
DEV-4961: c27 previous period gross outlay

### PR DESCRIPTION
**Pull request by alburde1**
**High level description:**
Adding rule C27 to ensure GrossOutlayByAward_CPE are carried through the periods in any given fiscal year

**Technical details:**
N/A

**Link to JIRA Ticket:**
[DEV-4961](https://federal-spending-transparency.atlassian.net/browse/DEV-4961)

The following are ALL required for the PR to be merged:
- [ ] Backend review completed
- [x] Unit & integration tests updated with relevant test cases
- Frontend impact assessment completed
- Documentation Updated

### CODE REVIEW

File: dataactvalidator/config/sqlrules/c27_award_financial.sql [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	po.uri,
+	po.piid,
+	po.parent_award_id,
+	po.gross_outlay_amount_by_awa_cpe,
+	po.tas AS "uniqueid_TAS",
+	po.disaster_emergency_fund_code AS "uniqueid_DisasterEmergencyFundCode",
+	po.fain AS "uniqueid_FAIN",
+	po.uri AS "uniqueid_URI",
+	po.piid AS "uniqueid_PIID",
+	po.parent_award_id AS "uniqueid_ParentAwardId"
```
Comment (dpb-bah) [inline dataactvalidator/config/sqlrules/c27_award_financial.sql:50]: Any reason why these are indented the way they are?

File: dataactvalidator/config/sqlrules/c27_award_financial.sql [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+FROM c27_prev_outlays_{0} AS po
+WHERE NOT EXISTS (
+	SELECT 1
+	FROM award_financial AS af
+	WHERE UPPER(COALESCE(po.fain, '')) = UPPER(COALESCE(af.fain, ''))
+		AND UPPER(COALESCE(po.uri, '')) = UPPER(COALESCE(af.uri, ''))
+		AND UPPER(COALESCE(po.piid, '')) = UPPER(COALESCE(af.piid, ''))
+		AND UPPER(COALESCE(po.parent_award_id, '')) = UPPER(COALESCE(af.parent_award_id, ''))
+		AND UPPER(COALESCE(po.tas, '')) = UPPER(COALESCE(af.tas, ''))
+		AND UPPER(COALESCE(disaster_emergency_fund_code, '')) = UPPER(COALESCE(disaster_emergency_fund_code, ''))
```
Comment (dpb-bah) [inline dataactvalidator/config/sqlrules/c27_award_financial.sql:60]: Really curious if how SQL knows which DEFC is which...

**Comment by dpb-bah** [comment]
Also per styleguide "- Indentations should be 4 spaces each (except when specifically aligning with an operator)"

File: dataactvalidator/config/sqlrules/c27_award_financial.sql (hunk 1 above)
Comment (alburde1) [reply dataactvalidator/config/sqlrules/c27_award_financial.sql:50]: Agh, looks like postico copy-paste messed everything up. Fixing

File: dataactvalidator/config/sqlrules/c27_award_financial.sql (hunk 2 above)
Comment (alburde1) [reply dataactvalidator/config/sqlrules/c27_award_financial.sql:60]: Weird that it didn't yell at me for that...

**Pull request merged**